### PR TITLE
generate master branch coverage

### PR DESCRIPTION
**Pull request by HarrisChu**
copy from `pull_request` workflow, just modify trigger.

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
Could place this job into the nightly workflow

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          res=$(git diff -U0 --no-color HEAD^ | /usr/share/clang/clang-format-10/clang-format-diff.py -p1)
+          [[ ! -z "$res" ]] && exit 1 || true
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+      - name: Prepare Gherkin exec environ
+        run: make init-all -C tests
+      - name: Check Gherkin feature format
+        run: make check -C tests
+
```
[the remark this answers is not in the document]
Comment (HarrisChu) [reply .github/workflows/coverage.yml:36]: it's OK for me.
nightly only run on `0 18 * * *`, I just want an additional job  at noon

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          gcc-*)
+            case ${{ matrix.os }} in
+            centos7)
+              # normal cluster
+              make up
+              ;;
+            ubuntu2004)
+              # ssl cluster
+              make ENABLE_SSL=true CA_SIGNED=true up
+              ;;
```
[the remark this answers is not in the document]
Comment (HarrisChu) [reply .github/workflows/coverage.yml:134]: cause it would show diff coverage, it's better to keep consistent with coverage in PR workflow.

**Review by yixinglu — APPROVED** [review]
LGTM

File: .github/workflows/coverage.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yixinglu) [reply .github/workflows/coverage.yml]: make sense.

**Comment by Shylock-Hg** [comment]
Why need this?

**Comment by HarrisChu** [comment]
> Why need this?

our pr workflow does not run coverage on master branch, just run coverage on commit. (i.e. just on pr, not on push).
if there's no base coverage, codecov could not trigger the pr comments.

**Comment by Shylock-Hg** [comment]
> > Why need this?
> 
> our pr workflow does not run coverage on master branch, just run coverage on commit. (i.e. just on pr, not on push).
> if there's no base coverage, codecov could not trigger the pr comments.

So, you could add the `on_push` event on master branch.

**Review by Shylock-Hg — APPROVED** [review]
LGTM

**Comment by HarrisChu** [comment]
> > > Why need this?
> > 
> > 
> > our pr workflow does not run coverage on master branch, just run coverage on commit. (i.e. just on pr, not on push).
> > if there's no base coverage, codecov could not trigger the pr comments.
> 
> So, you could add the `on_push` event on master branch. Nightly may out of date.

yes, but right now we have less runner, and it may block other prs.
I will follow this, if we have enough runner, change it to `on_push` event

**Pull request merged**